GithubBrowserSample/app/src/main/java/com/android/example/github/repository/NetworkBoundResource.kt

```diff
@@ -88,11 +88,9 @@ abstract class NetworkBoundResource<ResultType, RequestType>
                     }
                 }
                 is ApiEmptyResponse -> {
-                    appExecutors.mainThread().execute {
```

Comment:
There is also a switch to main thread here:

https://github.com/googlesamples/android-architecture-components/blob/523ea80ed325f637bca3ee6d36f56442ad4f5219/GithubBrowserSample/app/src/main/java/com/android/example/github/repository/NetworkBoundResource.kt#L80

Does LiveData automatically observe on the main thread?

Reply:
I found this and it looks like there is an assert that setValue is on the main thread:

https://gist.github.com/joseprl89/74e2fd256c214e70934b62bf3f9e2b58#file-mutablelivedatasetvalue-java-L2

Reply:
> There is also a switch to main thread here:
> 
> [android-architecture-components/GithubBrowserSample/app/src/main/java/com/android/example/github/repository/NetworkBoundResource.kt](https://github.com/googlesamples/android-architecture-components/blob/523ea80ed325f637bca3ee6d36f56442ad4f5219/GithubBrowserSample/app/src/main/java/com/android/example/github/repository/NetworkBoundResource.kt#L80)
> 
> Line 80 in [523ea80](/googlesamples/android-architecture-components/commit/523ea80ed325f637bca3ee6d36f56442ad4f5219)
> 
>  appExecutors.mainThread().execute { 
> Does LiveData automatically observe on the main thread?

https://github.com/googlesamples/android-architecture-components/blob/523ea80ed325f637bca3ee6d36f56442ad4f5219/GithubBrowserSample/app/src/main/java/com/android/example/github/repository/NetworkBoundResource.kt#L78
because there has switched to other thread to saveCallResult and processResponse, then we need switch to main thread to loadFromDb